tests/firedrake/conftest.py

```diff
@@ -10,6 +10,83 @@
 from firedrake.petsc import PETSc, get_external_packages
 
 
+def _skip_test_dependency(dependency):
+    """
+    Returns whether to skip tests with a certain dependency.
+
+    Usually, this will return True if the dependency is not available.
+    However, on CI we never want to skip tests because we should
+    test all functionality there, so if the environment variable
+    FIREDRAKE_CI=1 then this will always return False.
+    """
+    skip = True
```

Comment:
This reads quite strangely. Why not just have things like:
```py
if dependency == "slepc":
        try:
            from slepc4py import SLEPc  # noqa: F401
            del SLEPc
            return False
        except ImportError:
            return True
```
instead?

Reply:
This is related to your comment below. Because there is a switch for if we are on CI, we specifically are not asking "is this dependency present", but we are asking "should we skip tests with this dependency or not".

The `skip = True` line does read a bit strangely, but (IMO) it makes the lines lower down read more clearly. Because you would usually be asking if a dependency exists, `except ImportError: return True` is a bit odd at first glance, whereas `except ImportError: return skip` reads more like pseudocode.

Would adding a comment to the `skip = True` line be satisfactory?

Reply:
I see. You've convinced me. Happy to leave as is.